lib/Parse/ParseExpr.cpp

```diff
@@ -836,9 +836,11 @@ static StringRef copyAndStripUnderscores(ASTContext &C, StringRef orig) {
   char *start = static_cast<char*>(C.Allocate(orig.size(), 1));
   char *p = start;
```

Comment:
@jtbandes Yes, this issue is probably a bit theoretical but I'm trying to keep `clang-tidy` happy :-)